srcs/commands/join.cpp

```diff
@@ -70,17 +70,25 @@ void	join(Server *server, int const client_fd, cmd_struct cmd_infos)
 			cmd_infos.message.erase(cmd_infos.message.find(key), key.length()); // on erase la key de la string
 			if (key != it->second.getChannelPassword())
 			{
-				sendServerRpl(client_fd, ERR_BADCHANNELKEY(client_nickname, channel_name));
+				addToClientBuffer(server, client_fd, ERR_BADCHANNELKEY(client_nickname, channel_name));
 				continue; // on passe la suite, au prochain channel à ajouter síl y en a un
 			}
 		}
 
-		// vérifier si le client est banned avant de le join au channel
 		std::map<std::string, Channel>::iterator it_chan = server->getChannels().find(channel_name);
-		if (it_chan->second.isBanned(client_nickname) == true) {
+		// vérifier si le channel est full (METTRE UN PUTAIN DE RETURN avec l'error)
+		if (it_chan->second.getCapacityLimit() != -1 && (int)it_chan->second.getClientList().size() == it_chan->second.getCapacityLimit())
+		{
+			addToClientBuffer(server, client_fd, ERR_CHANNELISFULL(client_nickname, channel_name));
+			return ;
```

Comment:
```suggestion
			continue ;
```
will be modify like that